icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -3971,9 +3971,9 @@ SimpleDateFormat::setContext(UDisplayContext value, UErrorCode& status)
     if (U_SUCCESS(status)) {
         if ( fCapitalizationBrkIter == NULL && (value==UDISPCTX_CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE ||
                 value==UDISPCTX_CAPITALIZATION_FOR_UI_LIST_OR_MENU || value==UDISPCTX_CAPITALIZATION_FOR_STANDALONE) ) {
-            UErrorCode status = U_ZERO_ERROR;
-            fCapitalizationBrkIter = BreakIterator::createSentenceInstance(fLocale, status);
```

Comment:
For this one, I wonder if it might be better to use the status parameter that is passed in from the caller instead.
The reason is that createInstance could fail due to Out-of-memory, and the U_MEMORY_ALLOCATION_ERROR error would be hidden, and the caller would have no idea what an OOM error happened.

Reply:
Thanks Jeff, that's a good point.